src/js/providers/providers-supported.js

```diff
@@ -35,6 +35,7 @@ const SupportsMatrix = [
 
             var file = source.file;
             var type = source.type;
+            var mimeType = source.mimeType;
```

Comment:
Declare mimeType properly:

`const mimeType  = MimeTypes[type] || source.mimeType;`

Then you can simplify the rest of this method by just using the mime type as intended.